sdk/python/feast/expediagroup/vectordb/milvus_online_store.py

```diff
@@ -296,3 +305,96 @@ def _format_data_for_milvus(self, feast_data):
+        """
+
+        # Here we are constructing the feature list to request from Milvus with their relevant types
+
+        features_with_types = list(tuple())
+        for field in collection.schema.fields:
+            if field.name in features_to_request:
+                features_with_types.append(
+                    (field.name, self._milvus_to_feast_type(field.dtype))
+                )
+
+        feast_type_result = []
+        value_type_actions = {
+            "float_list_val": lambda val: ValueProto(float_list_val=FloatList(val=val)),
+            "string_val": lambda val: ValueProto(string_val=val),
+            "int64_val": lambda val: ValueProto(int64_val=val),
+            "double_val": lambda val: ValueProto(double_val=val),
+        }
+        prefix = "valuetype."
+
+        for row in milvus_result:
+            result_row = {}
+            for feature, feast_type in features_with_types:
+
+                value_proto = ValueProto()
+                feature_value = row[feature]
+
+                if feature_value:
+                    # Doing some pre-processing here to remove prefix
+                    value_type_method = f"{feast_type.to_value_type()}_val".lower()
+                    if value_type_method.startswith(prefix):
+                        value_type_method = value_type_method[len(prefix) :]
+                    # Construct the appropriate ValueProto instance.
+                    value_proto_constructor = value_type_actions.get(value_type_method)
+                    if value_proto_constructor:
+                        value_proto = value_proto_constructor(feature_value)
+
+                result_row[feature] = value_proto
+            # Append result after conversion to Feast Type
+            feast_type_result.append(result_row)
+        return feast_type_result
+
+    def _construct_milvus_query(self, entities) -> str:
+        """
+        Construct a Milvus query expression based on entity_keys provided.
+
+        Parameters:
+        entities (List[Entity]): List of entities with join keys and values.
+
+        Returns:
+        str: Constructed Milvus query expression.
+        """
+
+        milvus_query_expr = ""
+        entity_join_key = []
+        values_to_search = []
+
+        for entity in entities:
+            for key in entity.join_keys:
+                entity_join_key.append(key)
+            for value in entity.entity_values:
+                val_type = value.WhichOneof("val")
+                value_to_search = getattr(value, val_type)
+                values_to_search.append(value_to_search)
+
+        # TODO: Enable multiple join key support. Currently only supporting a single primary key/ join key
+        milvus_query_expr = f"{entity_join_key[0]} in {values_to_search}"
+
+        return milvus_query_expr
+
+    def _feast_to_milvus_data_type(self, feast_type) -> DataType:
+        mapped = FeastType(feast_type).name
```

Comment:
I know the enum was my suggestion, but I think the code has become too complicated for what it does. Can you check this library: https://pypi.org/project/bidict/ for bi-directional mapping instead?